File: Easy/1_Two_Sum/two_sum.go
package twosum

func twoSum(nums []int, target int) []int {
    storeMap := make(map[int]int)

    for index, value := range nums {
        num1 := value
        num2 := target - num1
        if index2,ok := storeMap[num2] ; ok {
            return []int{index,index2}
        }else{
            storeMap[num1] = index
        }
    }
    return []int{}
}

File: Easy/1_Two_Sum/two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        store_dic = {}
        for index, value in enumerate(nums):
            if (num2 := target - value) in store_dic:
                return [store_dic[num2],index]
            store_dic[value] = index
        return []
